extension/app/scripts.babel/pwa-check.js

```diff
@@ -151,20 +183,72 @@ var functionsToInject = [
   ['postAuditResults', postAuditResults]
 ];
 
-var audits = [
-  [hasManifest, 'Has a manifest'],
-  [isOnHTTPS, 'Site is on HTTPS'],
-  [hasCanonicalUrl, 'Site has a canonical URL'],
-  [hasManifestThemeColor, 'Site manifest has theme_color'],
-  [hasManifestBackgroundColor, 'Site manifest has background_color'],
-  [hasManifestStartUrl, 'Site manifest has start_url'],
-  [hasManifestShortName, 'Site manifest has short_name'],
-  [hasManifestName, 'Site manifest has name'],
-  [hasManifestIcons, 'Site manifest has icons defined'],
-  [hasManifestIcons192, 'Site manifest has 192px icon'],
-  [isControlledByServiceWorker, 'Site is currently controlled by a service worker'],
-  [hasServiceWorkerRegistration, 'Site has a service worker registration']
-];
+var audits = {
+  Security: [
+    [isOnHTTPS, 'Served over HTTPS']
+  ],
+  Offline: [
+    [hasServiceWorkerRegistration, 'Has a service worker registration']
+  ],
+  Manifest: [
+    [hasManifest, 'Exists'],
+    [hasManifestThemeColor, 'Contains theme_color'],
+    [hasManifestBackgroundColor, 'Contains background_color'],
+    [hasManifestStartUrl, 'Contains start_url'],
+    [hasManifestShortName, 'Contains short_name'],
+    [hasManifestName, 'Contains name'],
+    [hasManifestIcons, 'Contains icons defined'],
+    [hasManifestIcons192, 'Contains 192px icon'],
+  ],
+  Miscellaneous: [
+    [hasCanonicalUrl, 'Site has a Canonical URL']
+  ]
+};
+
+function createResultsHTML(results) {
+  const resultsGroup = {};
+  let groupName = null;
+
+  // Go through each group and restructure the results accordingly.
+  results.forEach(result => {
```

Comment:
forEach makes more sense here, since we're not reducing so much as refactoring the results.